### DISCUSSION TITLE
Registering additional icon sets with Mermaid

### DISCUSSION THREAD

**Discussion by jgosmann** [slidevjs/slidev#1943] [Q&A] [answered]
How do I register additional icon sets with Mermaid?

I tried to put the [code snippets from the Mermaid documentation](https://mermaid.js.org/config/icons.html) into `setup/main.ts` and `setup/mermaid.ts`, but neither did work.

I also had add this `vite.config.ts` so that I don't get an import error:
```
import { defineConfig } from 'vite';

export default defineConfig({
    optimizeDeps: {
        include: ['@braintree/sanitize-url', 'dayjs', 'debug']
    }
});
```

But I'm still seeing this instead of the desired icon:
![image](https://github.com/user-attachments/assets/7cac6def-9f5b-4ff2-a7e5-d6cd7f0e313f)

**Comment by jgosmann** [comment, marked as answer]
Well, the moment you post a question, you figure it out. Apparently it is important to use exactly the same import path [that is also used by Slidev](https://github.com/slidevjs/slidev/blob/688e9cf778421426992d391a1ab2de1c54dcfc9e/packages/client/modules/mermaid.ts#L3C22-L3C50).

With this code in `setup/main.ts` it works:

```ts
import { defineAppSetup } from '@slidev/types';
import mermaid from 'mermaid/dist/mermaid.esm.mjs';
import { icons } from '@iconify-json/logos';

mermaid.registerIconPacks([
    {
        name: icons.prefix,
        icons
    },
])

export default defineAppSetup(() => {})
```

**Reply by okelet** [reply]
This does not work for me. When the slide is shown, I get this error:

```
slide failed to load TypeError: (intermediate value).registerIconPacks is not a function
    <anonymous> mermaid.ts:5
```
